clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/cache/LoggingInstrumentation.groovy

```diff
@@ -33,7 +35,8 @@ class LoggingInstrumentation implements ExecutionInstrumentation {
 
   @Override
   void executionCompleted(Agent agent) {
-    logger.info("${agent.providerName}:${agent.agentType} completed")
+    def durationMs = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - MetricInstrumentation.getAgentStartTimeNs(agent))
```

Comment:
I'm not sure there is a solid guarantee that this executionCompletion is going to happen before the MetricInstrumentation executionCompletion (which would have cleared the startTime out of the ThreadLocal)
I'd be pretty ok with changing ExecutionInstrumentation callback to get passed the elapsed time and have the AgentScheduler runnable deal with tracking the elapsed time and killing off that whole dirty threadlocal